### ISSUE TITLE
~/.zprezto/modules/completion/init.zsh:37: parse error near `]]'

### ISSUE THREAD

**Issue by askurihin** [sorin-ionescu/prezto#1851]
### Description
After resent update (ff91c8d4), when I open a new shell, I recieve this error:

```shell
~/.zprezto/modules/completion/init.zsh:37: parse error near `]]'
```

### Expected behavior

I expect no errors during shell init.

### Steps to Reproduce

1. open zsh version 5.02
2. update prezto to ff91c8d4
3. open a new zsh session (type zsh)

### Versions

  - Prezto commit:
ff91c8d410df3e6141248474389051c7ddcaf80a
  - ZSH version:
zsh 5.0.2 (x86_64-redhat-linux-gnu)
  - OS information:
Centos 7.4

**Comment by srijanshetty** [comment]
`if [[ $_comp_path(#qNmh-20) ]];` uses `#q` which is supported only after version 5.06 as indicated in the [changelog](http://zsh.sourceforge.net/releases.html).
> If the option EXTENDED_GLOB is in effect, it is possible to force globbing within conditional code using the [[ ... ]] syntax by flagging that a certain string is a glob using the (#q) glob qualifier syntax. The resulting glob is treated as a single argument. For example, [[ -n *.c(#qN) ]] tests whether there are any .c files in the current directory.

**Comment by belak** [comment]
I don't know if we have an "officially supported versions" of zsh... I usually use Debian's oldstable as a benchmark (which is at 5.3.1). CentOS 7 (which is supported until some time in 2024 is on 5.0.2). CentOS 8 has 5.5.2.

Honestly, I don't like supporting CentOS because the package versions are so old. 5.0.2 came out at the end of 2012, almost 8 years ago. However, if this is the only thing preventing usage on 5.0.2, we could drop the `h` flag and tweak the numbers a bit.

**Comment by srijanshetty** [comment]
@belak, thanks for commenting. Probably we need consensus on official supported versions, or would that be a long drawn process?

The fix to remove `q` would be minor, did noodle some code locally and can push that if need be.

**Comment by okin** [comment]
@srijanshetty Would you mind sharing the patch that makes it work?

**Comment by srijanshetty** [comment]
Taking a step back, the regeneration logic of 20 hours isn't working for me. The last time the cache was created for me was on Jul 23.

![Screenshot from 2020-08-12 15-44-05](https://user-images.githubusercontent.com/1744347/90004210-c9f08b80-dc84-11ea-8fba-b1d4d892e185.png)

**Comment by jahagirdar** [comment]
Was this resolved?
I am forced to use centos7 on some machines for tool compability issues. and am seeing this issue.

**Comment by belak** [comment]
I don't believe so. It's not a big priority for me and I don't have any centos boxes to test on, but if you have a patch which works, you're welcome to submit it.
